### PR TITLE
KEYCLOAK-3491 Revise Scripting Support

### PR DESCRIPTION
**Pull request by thomasdarimont**
Refactored the scripting infrastructure and added documentation.
Added tests and an authenticator template in JavaScript for a quickstart.

### CODE REVIEW

File: themes/src/main/resources/theme/base/admin/resources/templates/kc-component-config.html [hunk 1]
```diff
@@ -33,7 +33,7 @@
         </div>
 
         <div class="col-md-6" data-ng-show="option.type == 'Script'">
-            <div ng-model="config[option.name][0]" placeholder="Enter your script..." ui-ace="{ useWrapMode: true, showGutter: true, theme:'github', mode: 'javascript'}">
+            <div ng-model="config[option.name]" placeholder="Enter your script..." ui-ace="{ useWrapMode: true, showGutter: true, theme:'github', mode: 'javascript'}">
```
Comment (thomasdarimont) [inline themes/src/main/resources/theme/base/admin/resources/templates/kc-component-config.html]: How can I get the default value of the option shown here? 
the default value contains an authenticator template for a quickstart.

File: services/src/main/resources/scripts/authenticator-template.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// import enum for error lookup
```
Comment (thomasdarimont) [inline services/src/main/resources/scripts/authenticator-template.js]: This is the default authenticator template that should be shown in the code editor for a quick start.

**Comment by thomasdarimont** [comment]
Will update the docks tomorrow - but I think the PR is now ready for review.

**Comment by thomasdarimont** [comment]
FYI Just rebased the PR on current master.

File: themes/src/main/resources/theme/base/admin/resources/templates/kc-component-config.html (hunk 1 above)
Comment (stianst) [inline themes/src/main/resources/theme/base/admin/resources/templates/kc-component-config.html]: Not sure - it's probably burried in server info somewhere

**Comment by stianst** [comment]
Did you add docs?

File: themes/src/main/resources/theme/base/admin/resources/templates/kc-component-config.html (hunk 1 above)
Comment (thomasdarimont) [inline themes/src/main/resources/theme/base/admin/resources/templates/kc-component-config.html]: I found it ... i needed to propagate the default values of the configuration properties within the `AuthenticationConfigCreateCtrl` in `realm.js`

**Comment by thomasdarimont** [comment]
Yes, but I wanted to wait for feedback first - here they are: 
https://github.com/keycloak/server_admin_guide/pull/40

**Comment by thomasdarimont** [comment]
... added some missing license headers.

**Pull request merged**